### ISSUE TITLE
improvements to format_parameters

### ISSUE THREAD

**Issue by DominiqueMakowski** [easystats/parameters#155]
For some more complex structures, it doesn't seem to work properly

```r
format_parameters(lm(Sepal.Length ~ Petal.Length + (Species / Sepal.Width), data = iris))
```
```
                        (Intercept)                         Petal.Length                    Speciesversicolor                     Speciesvirginica 
                       "(Intercept)"                       "Petal.Length"               "Species (versicolor)"                "Species (virginica)" 
           Speciessetosa:Sepal.Width        Speciesversicolor:Sepal.Width         Speciesvirginica:Sepal.Width 
       "Speciessetosa / Sepal.Width" "Species (versicolor) / Sepal.Width"  "Species (virginica) / Sepal.Width" 
```
```r
format_parameters(lm(Sepal.Length ~ Petal.Length + (Species / (Sepal.Width * Petal.Width)), data = iris))
```
```
                                         (Intercept)                                         Petal.Length                                    Speciesversicolor 
                                       "(Intercept)"                                       "Petal.Length"                               "Species (versicolor)" 
                                    Speciesvirginica                            Speciessetosa:Sepal.Width                        Speciesversicolor:Sepal.Width 
                               "Species (virginica)"                        "Speciessetosa / Sepal.Width"                 "Species (versicolor) / Sepal.Width" 
                        Speciesvirginica:Sepal.Width                            Speciessetosa:Petal.Width                        Speciesversicolor:Petal.Width 
                 "Species (virginica) / Sepal.Width"                        "Speciessetosa / Petal.Width"                 "Species (versicolor) / Petal.Width" 
                        Speciesvirginica:Petal.Width                Speciessetosa:Sepal.Width:Petal.Width            Speciesversicolor:Sepal.Width:Petal.Width 
                 "Species (virginica) / Petal.Width"        "(Speciessetosa * Sepal.Width) / Petal.Width" "(Species (versicolor) * Sepal.Width) / Petal.Width" 
            Speciesvirginica:Sepal.Width:Petal.Width 
 "(Species (virginica) * Sepal.Width) / Petal.Width" 
```

**Comment by strengejacke** [comment]
Cleaning parameter names is a mess... Maybe we should re-write this function from scratch, possibly using regex, and doing it step by step:

1. determine the variables and factors, and find their levels, and "prepare" parameters from factors (i.e. `Speciesvirginica`-> `Species (virginica)`
2. clean parameters
3. replace parameter names with new names from step 1
4. replace any interaction `:` with ` * `.
5. replace poly and I()^2 with better wordings...

But maybe the code is already doing this, is hard to follow the current code in that function ;-)

**Comment by DominiqueMakowski** [comment]
Yes, it's quite complicated, and the edge cases appear as potentially unlimited 😕 

The code currently in format_parameters is pretty much already doing that, but trying to stay away from regex: it relies on parameters_type, which tries to get the structure of each parameter. But it's not perfect either, being quite static and not very flexible... an overhaul might indeed be welcome, but still it should start with parameters_type rather than crazy regexing over the name

**Comment by strengejacke** [comment]
I fixed some issues when detecting splines (these were blank in the output): https://github.com/easystats/parameters/commit/7936f2f7b6e0ec7f35386fcecbfcc1577aeabeee

Now we should try to improve printing...

In particular, we may have "knots" or "df" defined in `ns()`, and maybe not when we have `bs()`...

``` r
library(splines)
library(parameters)
model <- lm(Sepal.Length ~ Petal.Length * bs(Petal.Width), data = iris)
model_parameters(model)
#> Parameter                       | Coefficient |   SE |         95% CI |     t |  df |      p
#> --------------------------------------------------------------------------------------------
#> (Intercept)                     |        4.63 | 0.49 | [ 3.66,  5.60] |  9.40 | 142 | < .001
#> Petal.Length                    |        0.14 | 0.35 | [-0.55,  0.83] |  0.41 | 142 | 0.686 
#> bs(Petal.Width)1                |        1.15 | 1.00 | [-0.83,  3.13] |  1.15 | 142 | 0.253 
#> bs(Petal.Width)2                |       -5.11 | 1.88 | [-8.82, -1.39] | -2.72 | 142 | 0.007 
#> bs(Petal.Width)3                |       -2.70 | 1.55 | [-5.77,  0.37] | -1.74 | 142 | 0.085 
#> Petal.Length * bs(Petal.Width)1 |        0.16 | 0.53 | [-0.89,  1.20] |  0.30 | 142 | 0.766 
#> Petal.Length * bs(Petal.Width)2 |        1.15 | 0.45 | [ 0.27,  2.04] |  2.58 | 142 | 0.011 
#> Petal.Length * bs(Petal.Width)3 |        0.68 | 0.43 | [-0.18,  1.54] |  1.57 | 142 | 0.118

model <- lm(Sepal.Length ~ Petal.Length * ns(Petal.Width, 4), data = iris)
model_parameters(model)
#> Parameter                          | Coefficient |   SE |         95% CI |     t |  df |      p
#> -----------------------------------------------------------------------------------------------
#> (Intercept)                        |        3.74 | 0.60 | [ 2.55,  4.93] |  6.20 | 140 | < .001
#> Petal.Length                       |        0.77 | 0.42 | [-0.07,  1.61] |  1.82 | 140 | 0.071 
#> ns(Petal.Width, 4)1                |        0.26 | 1.39 | [-2.50,  3.02] |  0.19 | 140 | 0.853 
#> ns(Petal.Width, 4)2                |       -4.38 | 1.22 | [-6.80, -1.96] | -3.58 | 140 | < .001
#> ns(Petal.Width, 4)3                |       -0.17 | 1.52 | [-3.17,  2.84] | -0.11 | 140 | 0.912 
#> ns(Petal.Width, 4)4                |       -2.60 | 1.59 | [-5.73,  0.54] | -1.64 | 140 | 0.104 
#> Petal.Length * ns(Petal.Width, 4)1 |       -0.13 | 0.51 | [-1.14,  0.89] | -0.25 | 140 | 0.806 
#> Petal.Length * ns(Petal.Width, 4)2 |        0.77 | 0.36 | [ 0.06,  1.48] |  2.15 | 140 | 0.033 
#> Petal.Length * ns(Petal.Width, 4)3 |       -0.94 | 0.88 | [-2.68,  0.80] | -1.07 | 140 | 0.288 
#> Petal.Length * ns(Petal.Width, 4)4 |        0.66 | 0.34 | [ 0.00,  1.32] |  1.97 | 140 | 0.051
```

<sup>Created on 2019-11-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by strengejacke** [comment]
well, for splines it was actually simpler than expected:

``` r
library(parameters)
library(splines)
model <- lm(Sepal.Length ~ Petal.Length * bs(Petal.Width), data = iris)
model_parameters(model)
#> Parameter                    | Coefficient |   SE |         95% CI |     t |  df |      p
#> -----------------------------------------------------------------------------------------
#> (Intercept)                  |        4.63 | 0.49 | [ 3.66,  5.60] |  9.40 | 142 | < .001
#> Petal.Length                 |        0.14 | 0.35 | [-0.55,  0.83] |  0.41 | 142 | 0.686 
#> bs [1st knot]                |        1.15 | 1.00 | [-0.83,  3.13] |  1.15 | 142 | 0.253 
#> bs [2nd knot]                |       -5.11 | 1.88 | [-8.82, -1.39] | -2.72 | 142 | 0.007 
#> bs [3rd knot]                |       -2.70 | 1.55 | [-5.77,  0.37] | -1.74 | 142 | 0.085 
#> Petal.Length * bs [1st knot] |        0.16 | 0.53 | [-0.89,  1.20] |  0.30 | 142 | 0.766 
#> Petal.Length * bs [2nd knot] |        1.15 | 0.45 | [ 0.27,  2.04] |  2.58 | 142 | 0.011 
#> Petal.Length * bs [3rd knot] |        0.68 | 0.43 | [-0.18,  1.54] |  1.57 | 142 | 0.118

model <- lm(Sepal.Length ~ Petal.Length * bs(Petal.Width, degree = 4), data = iris)
model_parameters(model)
#> Parameter                    | Coefficient |   SE |         95% CI |     t |  df |      p
#> -----------------------------------------------------------------------------------------
#> (Intercept)                  |        4.26 | 0.61 | [  3.05, 5.48] |  6.93 | 140 | < .001
#> Petal.Length                 |        0.40 | 0.43 | [ -0.46, 1.26] |  0.92 | 140 | 0.359 
#> bs [1st knot]                |        1.86 | 1.31 | [ -0.73, 4.45] |  1.42 | 140 | 0.157 
#> bs [2nd knot]                |       -1.34 | 3.25 | [ -7.77, 5.09] | -0.41 | 140 | 0.681 
#> bs [3rd knot]                |       -4.67 | 3.81 | [-12.20, 2.87] | -1.22 | 140 | 0.223 
#> bs [4th knot]                |       -2.21 | 2.26 | [ -6.67, 2.25] | -0.98 | 140 | 0.330 
#> Petal.Length * bs [1st knot] |       -0.63 | 0.86 | [ -2.32, 1.07] | -0.73 | 140 | 0.467 
#> Petal.Length * bs [2nd knot] |        0.61 | 0.62 | [ -0.62, 1.84] |  0.98 | 140 | 0.327 
#> Petal.Length * bs [3rd knot] |        0.77 | 0.73 | [ -0.68, 2.22] |  1.05 | 140 | 0.295 
#> Petal.Length * bs [4th knot] |        0.42 | 0.59 | [ -0.74, 1.58] |  0.71 | 140 | 0.479

model <- lm(Sepal.Length ~ Petal.Length * ns(Petal.Width, df = 3), data = iris)
model_parameters(model)
#> Parameter                    | Coefficient |   SE |         95% CI |     t |  df |      p
#> -----------------------------------------------------------------------------------------
#> (Intercept)                  |        4.63 | 0.53 | [ 3.59,  5.66] |  8.80 | 142 | < .001
#> Petal.Length                 |        0.21 | 0.37 | [-0.52,  0.94] |  0.58 | 142 | 0.566 
#> ns [1st knot]                |       -2.86 | 1.09 | [-5.01, -0.72] | -2.64 | 142 | 0.009 
#> ns [2nd knot]                |       -3.22 | 1.24 | [-5.67, -0.77] | -2.60 | 142 | 0.010 
#> ns [3rd knot]                |       -3.86 | 1.32 | [-6.48, -1.25] | -2.92 | 142 | 0.004 
#> Petal.Length * ns [1st knot] |        0.68 | 0.33 | [ 0.03,  1.32] |  2.07 | 142 | 0.040 
#> Petal.Length * ns [2nd knot] |        0.82 | 0.71 | [-0.58,  2.22] |  1.16 | 142 | 0.247 
#> Petal.Length * ns [3rd knot] |        0.74 | 0.29 | [ 0.17,  1.32] |  2.57 | 142 | 0.011
```

<sup>Created on 2019-11-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by strengejacke** [comment]
maybe we can also use **degree** instead of **knot** here.

**Comment by strengejacke** [comment]
Just writing the test-unit, is this intended?

``` r
library(parameters)
model <- lm(Sepal.Length ~ Species / Petal.Length * Sepal.Width, data = iris)
model_parameters(model)
#> Parameter                                           | Coefficient |   SE |          95% CI |     t |  df |     p
#> ----------------------------------------------------------------------------------------------------------------
#> (Intercept)                                         |       -1.37 | 3.91 | [ -9.10,  6.36] | -0.35 | 138 | 0.727
#> Species [versicolor]                                |        3.17 | 5.50 | [ -7.70, 14.04] |  0.58 | 138 | 0.565
#> Species [virginica]                                 |       -0.99 | 5.27 | [-11.42,  9.43] | -0.19 | 138 | 0.851
#> Sepal.Width                                         |        1.72 | 1.12 | [ -0.49,  3.94] |  1.54 | 138 | 0.126
#> Species [setosa] * Petal.Length                     |        2.88 | 2.75 | [ -2.56,  8.31] |  1.05 | 138 | 0.297
#> Species [versicolor] * Petal.Length                 |        0.81 | 0.91 | [ -0.99,  2.60] |  0.89 | 138 | 0.375
#> Species [virginica] * Petal.Length                  |        1.45 | 0.61 | [  0.24,  2.66] |  2.37 | 138 | 0.019
#> Species [versicolor] * Sepal.Width                  |       -1.36 | 1.86 | [ -5.03,  2.31] | -0.73 | 138 | 0.466
#> Species [virginica] * Sepal.Width                   |       -0.43 | 1.66 | [ -3.71,  2.85] | -0.26 | 138 | 0.797
#> (Species [setosa] * Petal.Length) * Sepal.Width     |       -0.74 | 0.79 | [ -2.30,  0.81] | -0.95 | 138 | 0.345
#> (Species [versicolor] * Petal.Length) * Sepal.Width |       -0.03 | 0.34 | [ -0.71,  0.65] | -0.08 | 138 | 0.936
#> (Species [virginica] * Petal.Length) * Sepal.Width  |       -0.18 | 0.21 | [ -0.59,  0.24] | -0.85 | 138 | 0.395

model <- lm(Sepal.Length ~ Species / (Petal.Length * Sepal.Width), data = iris)
model_parameters(model)
#> Parameter                                         | Coefficient |   SE |          95% CI |     t |  df |     p
#> --------------------------------------------------------------------------------------------------------------
#> (Intercept)                                       |       -1.37 | 3.91 | [ -9.10,  6.36] | -0.35 | 138 | 0.727
#> Species [versicolor]                              |        3.17 | 5.50 | [ -7.70, 14.04] |  0.58 | 138 | 0.565
#> Species [virginica]                               |       -0.99 | 5.27 | [-11.42,  9.43] | -0.19 | 138 | 0.851
#> Species [setosa] * Petal.Length                   |        2.88 | 2.75 | [ -2.56,  8.31] |  1.05 | 138 | 0.297
#> Species [versicolor] * Petal.Length               |        0.81 | 0.91 | [ -0.99,  2.60] |  0.89 | 138 | 0.375
#> Species [virginica] * Petal.Length                |        1.45 | 0.61 | [  0.24,  2.66] |  2.37 | 138 | 0.019
#> Species [setosa] * Sepal.Width                    |        1.72 | 1.12 | [ -0.49,  3.94] |  1.54 | 138 | 0.126
#> Species [versicolor] * Sepal.Width                |        0.37 | 1.48 | [ -2.56,  3.29] |  0.25 | 138 | 0.805
#> Species [virginica] * Sepal.Width                 |        1.30 | 1.22 | [ -1.12,  3.71] |  1.06 | 138 | 0.291
#> Species [setosa] * Petal.Length * Sepal.Width     |       -0.74 | 0.79 | [ -2.30,  0.81] | -0.95 | 138 | 0.345
#> Species [versicolor] * Petal.Length * Sepal.Width |       -0.03 | 0.34 | [ -0.71,  0.65] | -0.08 | 138 | 0.936
#> Species [virginica] * Petal.Length * Sepal.Width  |       -0.18 | 0.21 | [ -0.59,  0.24] | -0.85 | 138 | 0.395
```

<sup>Created on 2019-11-15 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by strengejacke** [comment]
I'm really not sure about all the details of `parameters_type()`, but after fixing some issues, I think we can actually keep the things as they are. I don't think there's the need for a new re-implementation right now, and I have added tests for many different use cases.

Not sure, however, how we want to have things like `log()` etc. behave - include in the output or not? Else, from the current situation, I think it's easy to extend the functions, as you only need little changes at few locations in the source (I think I have added relevant comments to these places...).

**Comment by strengejacke** [comment]
Can you elaborate more on your initial post? What is the current output, what is expected?

**Comment by strengejacke** [comment]
_bump_

**Comment by DominiqueMakowski** [comment]
So for simple nesting, for instance `y ~ group / x`, the "effect" of x will be estimated within each level of group (rather than showing the effect in the reference level and the change associated with changing that level).

In that case, parameters are:
- (Intercept)
- groupB 
- groupA / x
- groupB / x

Following that, in the case of `y ~ group / (x * z)`, effects of x, z and their interaction is altogether nested within group, with the following parameters:
- (Intercept)
- groupB 
- groupA / x
- groupB / x
- groupA / z
- groupB / z
- groupA / (x * z)
- groupB / (x * z)

The previous implementation prettified `Speciesvirginica:Sepal.Width:Petal.Width`  as 
 `(Species (virginica) * Sepal.Width) / Petal.Width` instead of  `Species (virginica) / (Sepal.Width * Petal.Width)`

But again, in the (current) version, there is only `*` (and nested effects and interactions are pretty much treated alike), so at least we reproduce the ugly names (mainly replacing `:` by ` * `) so we're sure that there is no error. 

Dealing with nested effects properly (and differentiating them from regular interactions via `/`) might be a challenge

**Comment by strengejacke** [comment]
Can you take close look at these outputs and tell me if formatting is correct like this?

``` r
library(parameters)
data(iris)

m1 <- lm(Sepal.Length ~ Species / Petal.Length, data = iris)
format_parameters(m1)
#>                           (Intercept)                     Speciesversicolor 
#>                         "(Intercept)"                "Species [versicolor]" 
#>                      Speciesvirginica            Speciessetosa:Petal.Length 
#>                 "Species [virginica]"     "Species [setosa] / Petal.Length" 
#>        Speciesversicolor:Petal.Length         Speciesvirginica:Petal.Length 
#> "Species [versicolor] / Petal.Length"  "Species [virginica] / Petal.Length"

model_parameters(m1)
#> Parameter                           | Coefficient |   SE |         95% CI |     t |  df |      p
#> ------------------------------------------------------------------------------------------------
#> (Intercept)                         |        4.21 | 0.41 | [ 3.41,  5.02] | 10.34 | 144 | < .001
#> Species [versicolor]                |       -1.81 | 0.60 | [-2.99, -0.62] | -3.02 | 144 | 0.003 
#> Species [virginica]                 |       -3.15 | 0.63 | [-4.41, -1.90] | -4.97 | 144 | < .001
#> Species [setosa] / Petal.Length     |        0.54 | 0.28 | [ 0.00,  1.09] |  1.96 | 144 | 0.052 
#> Species [versicolor] / Petal.Length |        0.83 | 0.10 | [ 0.63,  1.03] |  8.10 | 144 | < .001
#> Species [virginica] / Petal.Length  |        1.00 | 0.09 | [ 0.82,  1.17] | 11.43 | 144 | < .001


m2 <- lm(Sepal.Length ~ Petal.Length + (Species / Sepal.Width), data = iris)
format_parameters(m2)
#>                          (Intercept)                         Petal.Length 
#>                        "(Intercept)"                       "Petal.Length" 
#>                    Speciesversicolor                     Speciesvirginica 
#>               "Species [versicolor]"                "Species [virginica]" 
#>            Speciessetosa:Sepal.Width        Speciesversicolor:Sepal.Width 
#>     "Species [setosa] / Sepal.Width" "Species [versicolor] / Sepal.Width" 
#>         Speciesvirginica:Sepal.Width 
#>  "Species [virginica] / Sepal.Width"

model_parameters(m2)
#> Parameter                          | Coefficient |   SE |        95% CI |     t |  df |      p
#> ----------------------------------------------------------------------------------------------
#> (Intercept)                        |        1.67 | 0.41 | [ 0.87, 2.47] |  4.11 | 143 | < .001
#> Petal.Length                       |        0.82 | 0.07 | [ 0.69, 0.95] | 12.42 | 143 | < .001
#> Species [versicolor]               |        0.28 | 0.56 | [-0.82, 1.39] |  0.51 | 143 | 0.614 
#> Species [virginica]                |       -0.65 | 0.59 | [-1.81, 0.52] | -1.10 | 143 | 0.274 
#> Species [setosa] / Sepal.Width     |        0.62 | 0.12 | [ 0.40, 0.85] |  5.40 | 143 | < .001
#> Species [versicolor] / Sepal.Width |        0.18 | 0.15 | [-0.12, 0.47] |  1.17 | 143 | 0.245 
#> Species [virginica] / Sepal.Width  |        0.34 | 0.14 | [ 0.05, 0.62] |  2.36 | 143 | 0.020

m3 <- lm(Sepal.Length ~ Species / Petal.Length * Sepal.Width, data = iris)
format_parameters(m3)
#>                                           (Intercept) 
#>                                         "(Intercept)" 
#>                                     Speciesversicolor 
#>                                "Species [versicolor]" 
#>                                      Speciesvirginica 
#>                                 "Species [virginica]" 
#>                                           Sepal.Width 
#>                                         "Sepal.Width" 
#>                            Speciessetosa:Petal.Length 
#>                     "Species [setosa] / Petal.Length" 
#>                        Speciesversicolor:Petal.Length 
#>                 "Species [versicolor] / Petal.Length" 
#>                         Speciesvirginica:Petal.Length 
#>                  "Species [virginica] / Petal.Length" 
#>                         Speciesversicolor:Sepal.Width 
#>                  "Species [versicolor] * Sepal.Width" 
#>                          Speciesvirginica:Sepal.Width 
#>                   "Species [virginica] * Sepal.Width" 
#>                Speciessetosa:Petal.Length:Sepal.Width 
#>     "(Species [setosa] * Petal.Length) * Sepal.Width" 
#>            Speciesversicolor:Petal.Length:Sepal.Width 
#> "(Species [versicolor] * Petal.Length) * Sepal.Width" 
#>             Speciesvirginica:Petal.Length:Sepal.Width 
#>  "(Species [virginica] * Petal.Length) * Sepal.Width"

model_parameters(m3)
#> Parameter                                           | Coefficient |   SE |          95% CI |     t |  df |     p
#> ----------------------------------------------------------------------------------------------------------------
#> (Intercept)                                         |       -1.37 | 3.91 | [ -9.10,  6.36] | -0.35 | 138 | 0.727
#> Species [versicolor]                                |        3.17 | 5.50 | [ -7.70, 14.04] |  0.58 | 138 | 0.565
#> Species [virginica]                                 |       -0.99 | 5.27 | [-11.42,  9.43] | -0.19 | 138 | 0.851
#> Sepal.Width                                         |        1.72 | 1.12 | [ -0.49,  3.94] |  1.54 | 138 | 0.126
#> Species [setosa] / Petal.Length                     |        2.88 | 2.75 | [ -2.56,  8.31] |  1.05 | 138 | 0.297
#> Species [versicolor] / Petal.Length                 |        0.81 | 0.91 | [ -0.99,  2.60] |  0.89 | 138 | 0.375
#> Species [virginica] / Petal.Length                  |        1.45 | 0.61 | [  0.24,  2.66] |  2.37 | 138 | 0.019
#> Species [versicolor] * Sepal.Width                  |       -1.36 | 1.86 | [ -5.03,  2.31] | -0.73 | 138 | 0.466
#> Species [virginica] * Sepal.Width                   |       -0.43 | 1.66 | [ -3.71,  2.85] | -0.26 | 138 | 0.797
#> (Species [setosa] * Petal.Length) * Sepal.Width     |       -0.74 | 0.79 | [ -2.30,  0.81] | -0.95 | 138 | 0.345
#> (Species [versicolor] * Petal.Length) * Sepal.Width |       -0.03 | 0.34 | [ -0.71,  0.65] | -0.08 | 138 | 0.936
#> (Species [virginica] * Petal.Length) * Sepal.Width  |       -0.18 | 0.21 | [ -0.59,  0.24] | -0.85 | 138 | 0.395


m4 <- lm(Sepal.Length ~ Species / (Petal.Length * Sepal.Width), data = iris)
format_parameters(m4)
#>                                         (Intercept) 
#>                                       "(Intercept)" 
#>                                   Speciesversicolor 
#>                              "Species [versicolor]" 
#>                                    Speciesvirginica 
#>                               "Species [virginica]" 
#>                          Speciessetosa:Petal.Length 
#>                   "Species [setosa] / Petal.Length" 
#>                      Speciesversicolor:Petal.Length 
#>               "Species [versicolor] / Petal.Length" 
#>                       Speciesvirginica:Petal.Length 
#>                "Species [virginica] / Petal.Length" 
#>                           Speciessetosa:Sepal.Width 
#>                    "Species [setosa] / Sepal.Width" 
#>                       Speciesversicolor:Sepal.Width 
#>                "Species [versicolor] / Sepal.Width" 
#>                        Speciesvirginica:Sepal.Width 
#>                 "Species [virginica] / Sepal.Width" 
#>              Speciessetosa:Petal.Length:Sepal.Width 
#>     "Species [setosa] / Petal.Length / Sepal.Width" 
#>          Speciesversicolor:Petal.Length:Sepal.Width 
#> "Species [versicolor] / Petal.Length / Sepal.Width" 
#>           Speciesvirginica:Petal.Length:Sepal.Width 
#>  "Species [virginica] / Petal.Length / Sepal.Width"

model_parameters(m4)
#> Parameter                                         | Coefficient |   SE |          95% CI |     t |  df |     p
#> --------------------------------------------------------------------------------------------------------------
#> (Intercept)                                       |       -1.37 | 3.91 | [ -9.10,  6.36] | -0.35 | 138 | 0.727
#> Species [versicolor]                              |        3.17 | 5.50 | [ -7.70, 14.04] |  0.58 | 138 | 0.565
#> Species [virginica]                               |       -0.99 | 5.27 | [-11.42,  9.43] | -0.19 | 138 | 0.851
#> Species [setosa] / Petal.Length                   |        2.88 | 2.75 | [ -2.56,  8.31] |  1.05 | 138 | 0.297
#> Species [versicolor] / Petal.Length               |        0.81 | 0.91 | [ -0.99,  2.60] |  0.89 | 138 | 0.375
#> Species [virginica] / Petal.Length                |        1.45 | 0.61 | [  0.24,  2.66] |  2.37 | 138 | 0.019
#> Species [setosa] / Sepal.Width                    |        1.72 | 1.12 | [ -0.49,  3.94] |  1.54 | 138 | 0.126
#> Species [versicolor] / Sepal.Width                |        0.37 | 1.48 | [ -2.56,  3.29] |  0.25 | 138 | 0.805
#> Species [virginica] / Sepal.Width                 |        1.30 | 1.22 | [ -1.12,  3.71] |  1.06 | 138 | 0.291
#> Species [setosa] / Petal.Length / Sepal.Width     |       -0.74 | 0.79 | [ -2.30,  0.81] | -0.95 | 138 | 0.345
#> Species [versicolor] / Petal.Length / Sepal.Width |       -0.03 | 0.34 | [ -0.71,  0.65] | -0.08 | 138 | 0.936
#> Species [virginica] / Petal.Length / Sepal.Width  |       -0.18 | 0.21 | [ -0.59,  0.24] | -0.85 | 138 | 0.395


m5 <- lm(Sepal.Length ~ Petal.Length + (Species / (Sepal.Width * Petal.Width)), data = iris)
format_parameters(m5)
#>                                        (Intercept) 
#>                                      "(Intercept)" 
#>                                       Petal.Length 
#>                                     "Petal.Length" 
#>                                  Speciesversicolor 
#>                             "Species [versicolor]" 
#>                                   Speciesvirginica 
#>                              "Species [virginica]" 
#>                          Speciessetosa:Sepal.Width 
#>                   "Species [setosa] / Sepal.Width" 
#>                      Speciesversicolor:Sepal.Width 
#>               "Species [versicolor] / Sepal.Width" 
#>                       Speciesvirginica:Sepal.Width 
#>                "Species [virginica] / Sepal.Width" 
#>                          Speciessetosa:Petal.Width 
#>                   "Species [setosa] / Petal.Width" 
#>                      Speciesversicolor:Petal.Width 
#>               "Species [versicolor] / Petal.Width" 
#>                       Speciesvirginica:Petal.Width 
#>                "Species [virginica] / Petal.Width" 
#>              Speciessetosa:Sepal.Width:Petal.Width 
#>     "Species [setosa] / Sepal.Width / Petal.Width" 
#>          Speciesversicolor:Sepal.Width:Petal.Width 
#> "Species [versicolor] / Sepal.Width / Petal.Width" 
#>           Speciesvirginica:Sepal.Width:Petal.Width 
#>  "Species [virginica] / Sepal.Width / Petal.Width"

model_parameters(m5)
#> Parameter                                        | Coefficient |   SE |          95% CI |     t |  df |      p
#> --------------------------------------------------------------------------------------------------------------
#> (Intercept)                                      |        1.08 | 1.14 | [ -1.17,  3.33] |  0.95 | 137 | 0.344 
#> Petal.Length                                     |        0.87 | 0.07 | [  0.73,  1.01] | 11.92 | 137 | < .001
#> Species [versicolor]                             |       -0.73 | 2.64 | [ -5.95,  4.49] | -0.28 | 137 | 0.783 
#> Species [virginica]                              |       -5.35 | 3.10 | [-11.49,  0.79] | -1.72 | 137 | 0.087 
#> Species [setosa] / Sepal.Width                   |        0.78 | 0.33 | [  0.13,  1.43] |  2.36 | 137 | 0.019 
#> Species [versicolor] / Sepal.Width               |        0.99 | 0.90 | [ -0.79,  2.77] |  1.10 | 137 | 0.273 
#> Species [virginica] / Sepal.Width                |        2.16 | 1.00 | [  0.19,  4.12] |  2.16 | 137 | 0.032 
#> Species [setosa] / Petal.Width                   |        2.06 | 4.29 | [ -6.42, 10.54] |  0.48 | 137 | 0.632 
#> Species [versicolor] / Petal.Width               |        0.61 | 1.83 | [ -3.01,  4.22] |  0.33 | 137 | 0.740 
#> Species [virginica] / Petal.Width                |        2.44 | 1.44 | [ -0.40,  5.28] |  1.70 | 137 | 0.091 
#> Species [setosa] / Sepal.Width / Petal.Width     |       -0.61 | 1.22 | [ -3.01,  1.80] | -0.50 | 137 | 0.619 
#> Species [versicolor] / Sepal.Width / Petal.Width |       -0.45 | 0.66 | [ -1.75,  0.85] | -0.68 | 137 | 0.496 
#> Species [virginica] / Sepal.Width / Petal.Width  |       -0.88 | 0.48 | [ -1.83,  0.08] | -1.82 | 137 | 0.071
```

<sup>Created on 2019-11-18 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by DominiqueMakowski** [comment]
### m3

Up until m3 it seems that everything is correct! but I *think* that for e.g. `Speciessetosa:Petal.Length:Sepal.Width`, currently `(Species [setosa] * Petal.Length) * Sepal.Width`, should be `(Species [setosa] / Petal.Length) * Sepal.Width` (because the interaction impacts (`*:Sepal.Width`) impact what comes before (`Speciessetosa:Petal.Length`), which is `(Species [setosa] / Petal.Length)`

### m4

For `Speciesvirginica:Sepal.Width:Petal.Width`, currently `Species [virginica] / Sepal.Width / Petal.Width`, it should be `Species [virginica] / (Sepal.Width * Petal.Width)` because the whole interaction is nested within this level of species

**Comment by strengejacke** [comment]
This issue is related to https://github.com/easystats/parameters/issues/330, and I think we can close this now?
